Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Cancel.hs

```diff
@@ -103,6 +107,18 @@ cancel bookingId _ req = do
           Right res' -> do
             let merchantOperatingCityId = booking.merchantOperatingCityId
             disToPickup <- driverDistanceToPickup booking.merchantId merchantOperatingCityId (getCoordinates res'.currPoint) (getCoordinates booking.fromLocation)
+            now <- getCurrentTime
+            let driverBookingDuration = roundToIntegral $ diffUTCTime now booking.createdAt
+            distanceRes <-
+              Maps.getDistance booking.merchantId merchantOperatingCityId $
+                Maps.GetDistanceReq
+                  { origin = ride.driverInitialLocation,
+                    destination = getCoordinates res'.currPoint,
+                    travelMode = Just Maps.CAR
```

Comment:
Why extra map call required? Can we just check if `disToPickup <= estimatedDistance - 500` (where 500 is configurable)?